tck/features/WithAcceptance.feature

```diff
@@ -0,0 +1,361 @@
+      """
+      MATCH (a:A)
+      WITH a
+      ORDER BY a.name
+      LIMIT 1
+      MATCH (a)-->(b)
+      RETURN a
+      """
+    Then the result should be:
+      | a    |
+      | (:A) |
+    And no side effects
+
+  Scenario: No dependencies between the query parts
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A), (b:B)
+      """
+    When executing query:
+      """
+      MATCH (a)
+      WITH a
+      MATCH (b)
+      RETURN a, b
+      """
+    Then the result should be:
+      | a    | b    |
+      | (:A) | (:A) |
+      | (:A) | (:B) |
+      | (:B) | (:A) |
+      | (:B) | (:B) |
+    And no side effects
+
+  Scenario: Aliasing
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:Begin {prop: 42}),
+        (b:End {prop: 42}),
+        (c:End {prop: 3})
+      """
+    When executing query:
+      """
+      MATCH (a:Begin)
+      WITH a.prop AS property
+      MATCH (b:End)
+      WHERE property = b.prop
+      RETURN b
+      """
+    Then the result should be:
+      | b                 |
+      | (:End {prop: 42}) |
+    And no side effects
+
+  Scenario: Handle dependencies across WITH
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:End {prop: 42}),
+        (b:End {prop: 3}),
+        (c:Begin {prop: id(a)})
+      """
+    When executing query:
+      """
+      MATCH (a:Begin)
+      WITH a.prop AS property
+      LIMIT 1
+      MATCH (b)
+      WHERE id(b) = property
+      RETURN b
+      """
+    Then the result should be:
+      | b                 |
+      | (:End {prop: 42}) |
+    And no side effects
+
+  Scenario: Handle dependencies across WITH with SKIP
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a {prop: 'A', id: 0}),
+          (b {prop: 'B', id: id(a)}),
+          (c {prop: 'C', id: 0})
+      """
+    When executing query:
+      """
+      MATCH (a)
+      WITH a.prop AS property, a.id AS idToUse
+      ORDER BY property
+      SKIP 1
+      MATCH (b)
+      WHERE id(b) = idToUse
+      RETURN DISTINCT b
+      """
+    Then the result should be:
+      | b                    |
+      | ({prop: 'A', id: 0}) |
+    And no side effects
+
+  Scenario: WHERE after WITH should filter results
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a {name: 'A'}),
+        (b {name: 'B'}),
+        (c {name: 'C'})
+      """
+    When executing query:
+      """
+      MATCH (a)
+      WITH a
+      WHERE a.name = 'B'
+      RETURN a
+      """
+    Then the result should be:
+      | a             |
+      | ({name: 'B'}) |
+    And no side effects
+
+  Scenario: WHERE after WITH can filter on top of an aggregation
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a {name: 'A'}),
+        (b {name: 'B'})
+      CREATE (a)-[:REL]->()
+      CREATE (a)-[:REL]->()
+      CREATE (a)-[:REL]->()
+      CREATE (b)-[:REL]->()
```

Comment:
As above, we didn't decide on this, but I prefer using a new `CREATE` to distinguish between creating nodes and creating relationships, and to not repeat it if the pattern is very similar:

```
CREATE (a {name: 'A'}), (b {name: 'B'})
CREATE (a)-[:REL]->(),
       (a)-[:REL]->(),
       (a)-[:REL]->(),
       (b)-[:REL]->()
```

or line break with vertical alignment:

```
CREATE (a {name: 'A'}), 
       (b {name: 'B'})
CREATE (a)-[:REL]->(),
       (a)-[:REL]->(),
       (a)-[:REL]->(),
       (b)-[:REL]->()
```